processor/src/main/resources/org/mapstruct/ap/internal/model/IterableCreation.ftl

```diff
@@ -9,6 +9,8 @@
 <@compress single_line=true>
     <#if factoryMethod??>
         <@includeModel object=factoryMethod targetType=resultType/>
+    <#elseif enumSet>
+        EnumSet.noneOf( <@includeModel object=enumSetElementType raw=true/>.class )
```

Comment:
Is there somehow a way to do this more generic and don't hard-code the enum-set "instantiation" here in the template? Like the previous branch is there for all kind of factory methods.. or just keep it simple as we don't have any other use-case up to now?

Reply:
The reason why I did it like that was that it was already done in a similar way in [`ExistingInstanceSetterWrapperForCollectionsAndMaps.ftl`](https://github.com/mapstruct/mapstruct/blob/871353fccbf8ed05cfc7dee055f99eb81b0e69e0/processor/src/main/resources/org/mapstruct/ap/internal/model/assignment/ExistingInstanceSetterWrapperForCollectionsAndMaps.ftl#L39) and [`SetterWrapperForCollectionsAndMapsWithNullCheck.ftl`](https://github.com/mapstruct/mapstruct/blob/871353fccbf8ed05cfc7dee055f99eb81b0e69e0/processor/src/main/resources/org/mapstruct/ap/internal/model/assignment/SetterWrapperForCollectionsAndMapsWithNullCheck.ftl#L27).

I was thinking whether we can somehow create a fake factory method, but with the current way this was the most simple way to do it.

Reply:
Okay, then let us "keep it simple, stupid" as long as we don't need a more generic way suitable for similar cases :)